packages/cli/lib/ModuleManager.ts

```diff
@@ -0,0 +1,221 @@
+    }
+
+    return this._plugins.get(type);
+  }
+
+  async prepare() {
+    for (const application of this._applications.values()) {
+      if (application.prepare) {
+        await application.prepare();
+      }
+    }
+
+    for (const plugins of this.plugins.values()) {
+      for (const plugin of plugins.values()) {
+        if (plugin.prepare) {
+          await plugin.prepare();
+        }
+      }
+    }
+  }
+
+  async cleanup() {
+    for (const application of this._applications.values()) {
+      if (application.cleanup) {
+        await application.cleanup();
+      }
+    }
+
+    for (const plugins of this.plugins.values()) {
+      for (const plugin of plugins.values()) {
+        if (plugin.cleanup) {
+          await plugin.cleanup();
+        }
+      }
+    }
+  }
+
+  async configureApplicationCommands(yargs: Yargs.Argv) {
+    for (const application of this._applications.values()) {
+      const commandModules = await application.getCommandModules();
+      for (const commandModule of commandModules) {
+        // check requirements
+        if (!commandModule.command) {
+          throw new Error("SynTest command module required command property!");
+        } else if (!commandModule.describe) {
+          throw new Error("SynTest command module required describe property!");
+        } else if (!commandModule.handler) {
+          throw new Error("SynTest command module required handler function!");
+        }
+        yargs = yargs.command(commandModule);
+      }
+    }
+
+    return yargs;
+  }
+
+  async configurePluginOptions(yargs: Yargs.Argv) {
+    for (const plugins of this.plugins.values()) {
+      for (const plugin of plugins.values()) {
+        if (plugin.getOptions) {
+          const options = await plugin.getOptions();
+
+          for (const option of options.keys()) {
+            yargs = yargs.option(`${plugin.name}-${option}`, {
+              ...options.get(option),
+              group: `${plugin.name} ${options.get(option).group || ""}`,
+            });
+          }
+        }
+      }
+    }
+    return yargs;
+  }
+
+  async getModulePath(module: string): Promise<string> {
+    let modulePath = "";
+
+    if (module.startsWith("file:")) {
+      // It is a file path
+      modulePath = path.resolve(module.replace("file:", ""));
+      if (!shell.test("-e", modulePath)) {
+        throw new Error(`Filepath does not lead to an module, path: ${module}`);
+      }
+    } else {
+      // It is a npm package
+      modulePath = path.resolve(path.join("node_modules", module));
+
+      if (!shell.test("-e", modulePath)) {
+        // it is not locally installed lets try global
+        modulePath = path.resolve(path.join(globalModules, module));
+      }
+
+      if (!shell.test("-e", modulePath)) {
+        // it is not installed locally nor globally
+        // TODO maybe auto install?
+        throw new Error(
+          `Package is not installed locally or globally, package: ${module}`
```

Comment:
We should create a diagnostic entry for this

Reply:
Fixed.